source/PlanetLabel.cpp

```diff
@@ -24,65 +24,55 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include "Planet.h"
 #include "PointerShader.h"
 #include "Preferences.h"
+#include "Rectangle.h"
 #include "RingShader.h"
 #include "StellarObject.h"
 #include "System.h"
 #include "Wormhole.h"
 
 #include <algorithm>
 #include <cmath>
+#include <vector>
 
 using namespace std;
 
 namespace {
-	const double LINE_ANGLE[4] = {60., 120., 300., 240.};
-	const double LINE_LENGTH = 60.;
-	const double INNER_SPACE = 10.;
-	const double LINE_GAP = 1.7;
-	const double GAP = 6.;
-	const double MIN_DISTANCE = 30.;
-
-	// Check if the given label for the given stellar object and direction overlaps
-	// with any other stellar object in the system.
-	bool Overlaps(const System &system, const StellarObject &object, double zoom, double width, int direction)
+	// Label offset angles, in order of preference (non-negative only).
+	const vector<double> LINE_ANGLES = {60., 120., 300., 240., 30., 150., 330., 210., 90., 270., 0., 180.};
```

Comment:
It was originally a vector, I changed it to an array so it could be constexpr instead of const, then I looked around and found that we didn't use array anywhere, and it doesn't need to be constexpr, so I changed it back.

Reply:
> It was originally a vector,

[are you sure?](https://github.com/endless-sky/endless-sky/blob/master/source/PlanetLabel.cpp#L38) In any case, an array (whether std::array or C style) makes more sense :)

Reply:
Whoops, I mean, it was a static array originally.  I changed it to array so I could use ranges.  But, we don't use arrays, so I changed it to vector.

Reply:
just fyi you can use ranged for loops with C arrays too. We do use arrays (but pretty rarely that's true).

Reply:
Is there a reason not to use a vector?  I do like the ease of just being able to add new values without having to change anything else.

Didn't know that about ranges.

Reply:
Because it heap allocates. It's probably not really a big deal but yeah.  

>  I do like the ease of just being able to add new values without having to change anything else.

same. You can do

```suggestion
	constepr double LINE_ANGLES[] = {60., 120., 300., 240., 30., 150., 330., 210., 90., 270., 0., 180.};
```

Reply:
Huh.  So you can.